MidasCivil_Adapter/Convert/ToMidasCivil/Elements/FromFEMesh.cs

```diff
@@ -48,29 +53,70 @@ public static string FromFEMesh(this FEMesh feMesh)
+                    sectionPropertyId + "," +
+                  feMesh.Nodes[nodeIndices[0]].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                  feMesh.Nodes[nodeIndices[1]].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                  feMesh.Nodes[nodeIndices[2]].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                  feMesh.Nodes[nodeIndices[3]].AdapterId<string>(typeof(MidasCivilId)) + ",1,0");
+                }
+                else if (feMesh.Faces[0].NodeListIndices.Count == 3)
+                {
+                    midasElement = (feMesh.AdapterId<string>(typeof(MidasCivilId)) + ",PLATE," +
+                    materialId + "," +
+                    sectionPropertyId + "," +
+                 feMesh.Nodes[nodeIndices[0]].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                 feMesh.Nodes[nodeIndices[1]].AdapterId<string>(typeof(MidasCivilId)) + "," +
+                 feMesh.Nodes[nodeIndices[2]].AdapterId<string>(typeof(MidasCivilId)) + ",0,1,0");
+                }
             }
-            else
+            else if (feMesh.Faces.Count > 1)
             {
-                midasElement = (feMesh.AdapterId<string>(typeof(MidasCivilId)) + ",PLATE," +
-                materialId + "," +
-                sectionPropertyId + "," +
-             feMesh.Nodes[nodeIndices[0]].AdapterId<string>(typeof(MidasCivilId)) + "," +
-             feMesh.Nodes[nodeIndices[1]].AdapterId<string>(typeof(MidasCivilId)) + "," +
-             feMesh.Nodes[nodeIndices[2]].AdapterId<string>(typeof(MidasCivilId)) + ",0,1,0");
+                i++;
+                foreach (FEMeshFace meshFace in feMesh.Faces)
+                {
+                    int node0 = meshFace.NodeListIndices[0];
+                    int node1 = meshFace.NodeListIndices[1];
+                    int node2 = meshFace.NodeListIndices[2];
+           
+                    nodeIndices = meshFace.NodeListIndices;
+
+                    string j = i.ToString();//Will end up with random plate numbers
+                    FEMesh discreteMesh = new FEMesh();
+                    discreteMesh.Nodes = feMesh.Nodes;
+                    FEMeshFace disFace = new FEMeshFace();
+                    disFace.NodeListIndices.Add(node0);
+                    disFace.NodeListIndices.Add(node1);
+                    disFace.NodeListIndices.Add(node2);
+
+                    if (nodeIndices.Count == 4)
+                    {
+                        int node3 = meshFace.NodeListIndices[3];
+                        disFace.NodeListIndices.Add(node3);
+
+                    }
+                    discreteMesh.Faces.Add(disFace);
+
+                    discreteMesh.Property = feMesh.Property;
+                    discreteMesh.Property.Material = feMesh.Property.Material;
+                    
+
+                    discreteMesh.SetAdapterId<string>(typeof(MidasCivilId), j);
+                    midasElement += "\n" + FromFEMesh(discreteMesh);
+                    i++;
```

Comment:
Rather than rebuild each `Face` as a new `FEMesh` I think we can just make the entire loop from L57 to loop over each `Face` in the `FEMesh` and carry out L59-L81 creating an ELEMENT in MidasCivil.

Then all we need to do is add the method to work out the appropriate id (see comment above).

Reply:
Changes made